src/Agent.Worker/TestResults/TrxResultReader.cs

```diff
@@ -0,0 +1,448 @@
+
+                    XmlAttribute priorityAttribute = definitionNode.Attributes["priority"];
+                    if (priorityAttribute != null)
+                    {
+                        priority = priorityAttribute.Value;
+                    }
+
+                    XmlNode ownerNode = definitionNode.SelectSingleNode("./Owners/Owner");
+                    if (ownerNode != null)
+                    {
+                        IdentityRef ownerIdRef = new IdentityRef();
+                        ownerIdRef.DisplayName = ownerNode.Attributes["name"].Value;
+                        owner = ownerIdRef;
+                    }
+
+                    // The automated test name should be FQDN, if we are unable to figure it out like in case of a webtest, 
+                    // set it as "name" from the parent (where it is always present) 
+                    XmlNode testResultNode = definitionNode.SelectSingleNode("./TestMethod");
+                    string automatedTestName = null;
+                    if (testResultNode != null && testResultNode.Attributes["className"] != null && testResultNode.Attributes["name"] != null)
+                    {
+                        // At times the class names are coming as 
+                        // className="MS.TF.Test.AgileX.VSTests.WiLinking.UI.WiLinkingUIQueryTests"
+                        // at other times, they are as 
+                        // className="UnitTestProject3.UnitTest1, UnitTestProject3, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" 
+                        string className = testResultNode.Attributes["className"].Value.Split(',')[0];
+                        automatedTestName = className + "." + testResultNode.Attributes["name"].Value;
+                    }
+                    else if (definitionNode.Attributes["name"] != null)
+                    {
+                        automatedTestName = definitionNode.Attributes["name"].Value;
+                    }
+
+                    _definitions.Add(definitionNode.Attributes["id"].Value, new TestCaseDefinition(automatedTestName, owner, priority, storage));
+                }
+            }
+
+            // Read UnitTestResults as well as WebTestResults
+            XmlNodeList resultsNodes = doc.SelectNodes("/TestRun/Results/UnitTestResult");
+            XmlNodeList webTestResultNodes = doc.SelectNodes("/TestRun/Results/WebTestResult");
+
+            results.AddRange(ReadActualResults(resultsNodes, "UnitTest"));
+            results.AddRange(ReadActualResults(webTestResultNodes, "WebTest"));
+
+            testRunData.Results = results.ToArray<TestCaseResultData>();
+            _executionContext.Debug(string.Format(CultureInfo.InvariantCulture, "Total test results: {0}", testRunData.Results.Length));
+
+            return testRunData;
+        }
+
+        public bool AddResultsFileToRunLevelAttachments
+        {
+            get
+            {
+                return _addResultsFileToRunLevelAttachments;
```

Comment:
You could just use the property itself right? why do you need another private variable? Its not a complex object also, its just bool.